application/src/main/java/org/opentripplanner/model/plan/paging/cursor/PageCursorFactory.java

```diff
@@ -6,8 +6,10 @@
 import java.time.Duration;
 import java.time.Instant;
 import javax.annotation.Nullable;
+import org.opentripplanner.framework.model.Cost;
 import org.opentripplanner.model.plan.ItinerarySortKey;
 import org.opentripplanner.model.plan.SortOrder;
+import org.opentripplanner.routing.algorithm.filterchain.paging.DefaultPageCursorInput;
```

Comment:
This still have a dependency to filterchain.

Reply:
I removed this and modified logic related to null checking in the `PageCursorFactory` and `PagingService`. I noticed that the `assertRequestPrerequisites` function in `PagingService` is useless because of a preceding check. Should I remove it?